erts/emulator/asmjit/core/jitallocator.cpp

```diff
@@ -521,14 +521,17 @@ static Error JitAllocatorImpl_newBlock(JitAllocatorPrivateImpl* impl, JitAllocat
   if (block != nullptr)
     bitWords = static_cast<BitWord*>(::malloc(size_t(numBitWords) * 2 * sizeof(BitWord)));
 
+  VirtMem::MemoryFlags memoryFlags = VirtMem::MemoryFlags::kAccessRWX;
+  if (Support::test(impl->options, JitAllocatorOptions::kUseLargePages))
```

Comment:
This differs from `asmjit` upstream and will get clobbered as soon as we update it. In the PR you made upstream, the maintainer [asked you](https://github.com/asmjit/asmjit/pull/378#issuecomment-1712799408) to try their version of large pages but you didn't respond (in public at least).

Are there any problems with their implementation?

Reply:
Yes, the implementation does not work.

This is an unfortunate situation.  I contributed the change a year ago and the `asmjit` maintainer and I worked on adjusting the pull request to his liking with various functional and cosmetic changes.  Alas, in the end he would not accept it on the master branch.  Three months ago he picked out a few parts of my change and rewrote them for whatever reason without preserving the original behavior.

I am not sure why things have gone in this direction.   I can follow-up on the pull request and see if he can share his intentions for my pull request.

Reply:
> I am not sure why things have gone in this direction. I can follow-up on the pull request and see if he can share his intentions for my pull request.

I think that'd be a great help, I get the feeling you've been talking past each other.